samples/bootloader/README.rst

```diff
@@ -0,0 +1,155 @@
+.. _bootloader:
+
+Immutable bootloader
+####################
+
+The bootloader sample implements an immutable first stage bootloader that has the capability to verify and boot a second stage bootloader.
```

Comment:
Note that for some constrained devices, it is technically possible to boot the application directly from the immutable bootloader. This will give the secure boot property, but without support for device firmware upgrade.

Reply:
But it will complicate the docs a lot if we have to explain this in all scenarios. So I would just leave it as is, or have a small comment here, and don't mention it again.

Reply:
This is how I understood it - we're saying that it has the capability to boot a second stage bootloader, and that is has support for a second stage bootloader, but not that the next boot stage must be a bootloader. :)
Step 2 (Select the next stage in the boot chain) is a bit too specific though. I'll change that.

Reply:
Updated.